File: framework/web/src/main/java/top/bettercode/simpleframework/web/serializer/annotation/JsonKilogram.java
package top.bettercode.simpleframework.web.serializer.annotation;

import com.fasterxml.jackson.annotation.JacksonAnnotationsInside;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import top.bettercode.simpleframework.web.serializer.KilogramDeserializer;
import top.bettercode.simpleframework.web.serializer.KilogramSerializer;

/**
 * @author Peter Wu
 */
@Target({ElementType.METHOD, ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
@JacksonAnnotationsInside
@JsonSerialize(using = KilogramSerializer.class)
@JsonDeserialize(using = KilogramDeserializer.class)
public @interface JsonKilogram {

}

File: web/src/main/kotlin/top/bettercode/summer/web/serializer/annotation/JsonKilogram.kt
package top.bettercode.summer.web.serializer.annotation

import com.fasterxml.jackson.annotation.JacksonAnnotationsInside
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import top.bettercode.summer.web.serializer.KilogramDeserializer
import top.bettercode.summer.web.serializer.KilogramSerializer

/**
 * @author Peter Wu
 */
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
@JacksonAnnotationsInside
@JsonSerialize(using = KilogramSerializer::class)
@JsonDeserialize(using = KilogramDeserializer::class)
annotation class JsonKilogram
